### ISSUE TITLE
wrapReducerWithFormStateUpdate will break if the state is flat

### ISSUE THREAD

**Issue by abannsunny** [MrWolfZ/ngrx-forms#196]
I have used the example you have given for the wrapReducerWithFormStateUpdate function, to illustrate the issue.

When the state in the reducer is flat

So instead of a state like this:
```
const rawReducer = createReducer(
  {
    loginForm: createFormGroupState('loginForm', initialLoginFormValue),
    // your other properties...
  },
  onNgrxForms(),
  // your other reducers...
);
```

My state is: 

```
const rawReducer = createReducer(
createFormGroupState('loginForm', initialLoginFormValue),
  onNgrxForms(),
  // your other reducers...
);
```

This is from you example for the wrapReducerWithFormStateUpdate:

```
import { createReducer } from '@ngrx/store';
import { onNgrxForms, updateGroup, validate, wrapReducerWithFormStateUpdate } from 'ngrx-forms';
import { required } from 'ngrx-forms/validation';

export interface LoginFormValue {
  username: string;
  password: string;
  stayLoggedIn: boolean;
}

export const initialLoginFormValue: LoginFormValue = {
  username: '',
  password: '',
  stayLoggedIn: false,
};

export const validateLoginForm = updateGroup<LoginFormValue>({
  username: validate(required),
  password: validate(required),
});

const rawReducer = createReducer( createFormGroupState('loginForm', initialLoginFormValue),,
  onNgrxForms(),
  // your other reducers...
);

// wrapReducerWithFormStateUpdate calls the update function
// after the given reducer; you can wrap this reducer again
// if you have multiple forms in your state
export const reducer = wrapReducerWithFormStateUpdate(
  rawReducer,
  // point to the form state to update
  s => s,
  // this function is always called after the reducer
  validateLoginForm,
);
```
The formStateLocator in this case is just **s => s**.

In this case wrapReducerWithFormStateUpdate function does not update the state appropriately since 

    const formStateKey = Object.keys(updatedState).find(key => updatedState[key as keyof TState] as any === formState)!;

TFormState and TState are equal in this case and it cannot find the key.

So the return statement from the function:
```
    return {
      ...updatedState,
      [formStateKey]: updatedFormState,
    };
```

statement gives me a state like

```
{
originalState: originalValue,
undefined: modifiedValue
}
```
I did something like this for now to work around:(not sure if this the right approach)

```
export function wrapReducerWithFormStateUpdate<TState>(
  reducer: ActionReducer<TState>,
  updateFn: (state: TState) => TState,
): ActionReducer<TState> {
  return (state, action) => {
    const updatedState = reducer(state, action);

    const updatedFormState = updateFn(updatedState);

    if (updatedFormState === updatedState ) {
      return updatedState;
    }

    return {
      ...updatedState,
      ...updatedFormState,
    };
  };
}
```

**Comment by dzonatan** [comment]
As you pointed out yourself - `wrapReducerWithFormStateUpdate` is intended to use with states that have nested structure.
Instead of 
```ts
export const reducer = wrapReducerWithFormStateUpdate(
  rawReducer,
  // point to the form state to update
  s => s,
  // this function is always called after the reducer
  validateLoginForm,
);
```

use `createFormStateReducerWithUpdate`:
```ts
export const reducer = createFormStateReducerWithUpdate<LoginFormValue>(validateLoginForm)
```

or just simply wrap the `validateLoginForm` around your `rawReducer`:
```ts
export const reducer = (state: LoginFormValue, action: Action) => validateLoginForm(rawReducer(state, action));
```

**Comment by abannsunny** [comment]
I see, I did not know it was intended to be used in the case of nested structures.

Thanks you for the suggestions! Those look good to me.

**Comment by MrWolfZ** [comment]
Good point, `wrapReducerWithFormStateUpdate` should either just work or give a meaningful error message when used on form states directly. I'll update the code.

**Comment by MrWolfZ** [comment]
I've published the fix as version [6.3.3](https://github.com/MrWolfZ/ngrx-forms/releases/tag/v6.3.3).
